### ISSUE TITLE
kl_div formula in documentation

### ISSUE THREAD

**Issue by ThyrixYang** [pytorch/pytorch#32278]
## 📚 Documentation

I found that neither https://pytorch.org/docs/master/nn.functional.html#torch.nn.functional.kl_div

nor https://pytorch.org/docs/master/nn.html#torch.nn.KLDivLoss provide a formula of kl divergence to indicate what is the relationship of input, target and P, Q in wikipedia reference.

The kl_div(input, target) is actually D(target, input) as mentioned in https://discuss.pytorch.org/t/custom-loss-kl-divergence-error/19850/4?u=thyrixyang

Although in wikipedia it said D(p, q) is distance **of Q from P**, which suggests P should be the target, I think there should be a formula to make it clear. Otherwise, someone not very familiar with kl divergence like me may easily think kl_div(input, target)=D(input, target).

**Comment by SsnL** [comment]
<img width="799" alt="Screenshot 2020-01-16 11 54 33" src="https://user-images.githubusercontent.com/5674597/72545407-f9f2c000-3856-11ea-8dd2-121ed1869631.png">
The link you refer to does have a formula. Maybe it is not clear enough. How would you suggest improving it?

**Comment by ThyrixYang** [comment]
Hi, @SsnL ,

The kl_div loss seems the only loss that gives a reference link, which indicates it's relatively complex.  But when we go to wikipedia it's not clear at the first glance what is the relationship of P, Q, x, y, input , target... in all these formulae. The formula in the doc is not kl_div, neither mean nor sum reduction do the right "kl_div". The "correct KL divergence" "batchmean"  is only mentioned at the very last, which seems strange. It's hard to know what exactly is the "kl_div" if we only read the doc.

I suggest 

1. Remove the reference link if we are not going to use the same symbols as wikipedia, and add a formal definition of kl divergence in the doc, which is consistent with other loss and self-contained.  The wikipedia link is redundant since it's easy to google "kl divergence" and it's not consistent with our doc.

or

2. Describe the relationship of x, y and P, Q in wikipedia, and the relationship of l(x, y) and D_{kl}(P||Q) in wikipedia.

3. The batchmean deserves a formula than sum and mean and it should be introduced at first and highlighted.

**Comment by vadimkantorov** [comment]
Completely agreed. the terminology should be standard (P, Q), the names for inputs should be kept minimal and not ("input", "target", "x", "y" in different parts, given that they're all very non-standard compared to P and Q), the formula should be written in the standard formulation (like in wikipedia).

Another problem is that the explanation uses three different letters L: "big L", "small L", "small calligraphic L". For a basic function like this, maybe this could be streamlined and simplified as well.

**Comment by tqbl** [comment]
Note that this was resolved in #67443. This thread can be closed.

**Comment by vadimkantorov** [comment]
@Lezcano seems not very fixed for me. Yes, docs now include a warning plate explaining how input/target map to P and Q, but it may be much better to make this "renaming" in the beginning and explain the formula and functioning in familiar P/Q terms...

**Comment by Lezcano** [comment]
P, Q, although moderately standard, do not denote which one is the real distribution and which one is the predicted distribution. Hence the chosen notation in the docs.

This issue was resolved as `kl_div` does have the formula in the docs. If you still think that there's room for improvement (fair enough), feel free to open a new issue, or even better, submit a PR.
